scipy/special/specfun/specfun.f

```diff
@@ -1814,7 +1860,7 @@ SUBROUTINE REFINE(KD,M,Q,A)
         DELTA=0.0D0
         X0=A
         CALL CVF(KD,M,Q,X0,MJ,F0)
-        X1=1.002*A
+        X1=(1 + 1.0D-13)*A
```

Comment:
Secant method requires the roots to be close to the desired value (here `A`). I observed that `1.002*A` results in more discontinuities as compared to  `(1 + 1.0D-13)*A`.

Reply:
It seems like `REFINE` is very chaotic for this use case. A slight change in `x` in `(1 + x)*A` results in significant difference in results for large `q` and `n` values. Other methods depend significantly on some conditions to be met like two guessed roots should be of opposite signs, etc. Is it necessary to follow this approach? Can we use some other implementation/algorithm for computing the values of mathieu functions as the current one is unstable because of its dependence on iterative root finding algorithms.